src/HotChocolate/Caching/src/Caching/Types/CacheControlScope.cs

```diff
@@ -5,17 +5,17 @@ namespace HotChocolate.Caching;
 /// </summary>
 public enum CacheControlScope
 {
-    /// <summary>
-    /// Private query results contain data
-    /// that is scoped to a user and must therefore
-    /// be only cached for and accessible to that user.
-    /// </summary>
-    Private,
-
     /// <summary>
     /// Public query results contain data
     /// that is not scoped to a particular user and can
     /// therefore be stored in a shared cache.
     /// </summary>
     Public,
+
+    /// <summary>
+    /// Private query results contain data
+    /// that is scoped to a user and must therefore
+    /// be only cached for and accessible to that user.
+    /// </summary>
+    Private,
```

Comment:
The problem is that parsing a directive currently always initializes the properties of the runtime directive type, EVEN if they are nullable.
This is just a temporary fix that also causes other caching functionality to fail. I'm still investigating the underlying issue.